src/strife/r_things.c

```diff
@@ -812,6 +836,40 @@ void R_DrawPSprite (pspdef_t* psp)
         vis->colormap = colormaps + INVERSECOLORMAP * 256 * sizeof(lighttable_t);
     }
 
+    // [crispy] interpolate weapon bobbing
```

Comment:
The weapon bobbing interpolation shared among the different games doesn't work with certain weapons. This includes the chainsaw in Doom and the flamethrower and crossbow (electric bolts) in Strife and probably others. There may be a way to safely fix this in a future PR. [This Doomworld post has two solutions](https://www.doomworld.com/forum/topic/121183-which-of-these-chainsaw-animations-is-the-correct-one/?tab=comments#comment-2292505).

Reply:
I'd prefer the first solution, i.e. interpolate for each tic even if the weapon state isn't actually updated. This is done already in Crispy Doom if one of the "Weapon Attack Alignment" options is selected.

Reply:
Is it really as simple as this for Doom?

```diff
- if (psp->state && (crispy->bobfactor || crispy->centerweapon))
+ if (psp->state)
```

Reply:
> Is it really as simple as this for Doom?

Well, this would apply bobbing to any valid weapon sprite state - which means simply all in Vanilla Doom. 

> Edit: this preserves vanilla choppiness:

Yes, this looks like a reasonable solution.

Reply:
Strife weapon bobbing interpolation partially fixed by https://github.com/fabiangreffrath/crispy-doom/pull/998/commits/29eea2cd98493e986b43ee5ef38479db803a7221. It still isn't perfect and doesn't work on the crossbow (electric bolts) but this can be looked at later.